### PR TITLE
Update cuda arch and change cuda version

### PR DESCRIPTION
**Pull request by aurianer**
<s>- Downgrade the cuda version on rostam to avoid the errors currently on master
https://cdash.cscs.ch/viewBuildError.php?buildid=172746
- Update at the same time the cuda architecture as the sc15 nodes seems to have V100 now

Created this ticket #5472 to fix the problem with cuda/11.3</s>

EDIT:
- Workaround another bug in using noexcept with variadic templates in CUDA 11.4, @msimberg confirmed that it compiles with CUDA 11.3
- Update at the same time the cuda architecture as the sc15 nodes seems to have V100 now
- Pin CUDA version to 11.4 to avoid breaking master after a rostam default cuda package update
- Workaround bug in mismatch between pointers type which appeared with CUDA 11.3 and CUDA 11.4 but not in older versions

### CODE REVIEW

**Comment by aurianer** [comment]
launch perftests

**Comment by aurianer** [comment]
Looks like it only solves the `gcc-9-cuda-11-release` build

**Review by hkaiser — DISMISSED** [review]
LGTM, thanks!

**Comment by aurianer** [comment]
retest lsu

File: .jenkins/lsu/slurm-configuration-gcc-9-cuda-11.sh [hunk 1]
```diff
@@ -4,5 +4,5 @@
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-configuration_slurm_partition="sc15"
+configuration_slurm_nodes="geev"
```
Comment (msimberg) [inline .jenkins/lsu/slurm-configuration-gcc-9-cuda-11.sh:7]: Do we need this *if* we can compile for multiple architectures at the same time? It'd be nice to keep two nodes for CUDA builds if possible.

Comment (aurianer) [reply .jenkins/lsu/slurm-configuration-gcc-9-cuda-11.sh:7]: not at all that's why I called the commit [Tmp] I was just trying it out as I couldn't get a manual allocation on rostam

**Comment by msimberg** [comment]
retest lsu

**Comment by msimberg** [comment]
retest lsu

**Comment by msimberg** [comment]
Thanks @aurianer for updating this! It looks good except that there were odd failures on the release build with nvcc (executables not found even though they were built). My guess is that it was unfortunate timing with the maintenance on rostam, but let's see. I've retriggered the builds on rostam.

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Comment by aurianer** [comment]
Still have to investigate a compilation error which triggers the CI error (executables do not exist) so not ready to merge

**Comment by aurianer** [comment]
So I downgraded temporarily the CUDA version to 11.2 and will investigate the error which appeared for CUDA 11.3 and CUDA 11.4 and fix it in another PR

**Comment by msimberg** [comment]
retest lsu

File: libs/full/compute_cuda/include/hpx/compute/cuda/detail/launch.hpp [hunk 2]
```diff
@@ -109,6 +107,7 @@ namespace hpx { namespace cuda { namespace experimental { namespace detail {
                     std::string("kernel launch failed: ") +
                         cudaGetErrorString(error));
             }
+            HPX_UNUSED(launcher);
```
Comment (msimberg) [inline libs/full/compute_cuda/include/hpx/compute/cuda/detail/launch.hpp:110]: What's the reasoning to duplicate this in both branches?

Comment (aurianer) [reply libs/full/compute_cuda/include/hpx/compute/cuda/detail/launch.hpp:110]: Cause in one of them it's used, I was finding weird that we declared it unused and then used it ^^ but I don't mind reverting the change either

Comment (msimberg) [reply libs/full/compute_cuda/include/hpx/compute/cuda/detail/launch.hpp:110]: Ah, makes sense. I missed that there are three branches.

**Review by msimberg — APPROVED** [review]
This looks good to me, and works nicely on rostam!

I'd just like to wait for the CUDA configuration on CSCS to verify that it doesn't break with older CUDA versions. Piz Daint maintenance is almost over and should be available again later today, but it may first require changes to the CUDA configuration on Piz Daint.

**Comment by msimberg** [comment]
retest cscs

**Pull request merged**